packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIteratorItem.tsx

```diff
@@ -83,8 +83,14 @@ export const SimpleFormIteratorItem = React.forwardRef(
         const parentSourceContext = useSourceContext();
         const sourceContext = useMemo(
             () => ({
-                getSource: (source: string) =>
-                    source ? `${member}.${source}` : member,
+                getSource: (source: string) => {
+                    if (parentSourceContext) {
```

Comment:
There was a bug in nested ArrayInput, visible in the inputs / ArrayInput / NestedInline story